### ISSUE TITLE
[Question] MorphMany issue 

### ISSUE THREAD

**Issue by umarakber** [Laravel-Backpack/CRUD#1926]
Hi! How can i use polymorph with backpack ?

I have 2 models: ticket and todo
And i have 1 model "image" that have relations with "ticket" and "todo"
"ticket" have relation with "Image":

```
public function images()
    {
        return $this->morphMany('App\Models\Image', 'imageable');
    }

```
"Image" have relation with "ticket":
```

public function imageable()
    {
        return $this->morphTo();
    }
```

I created Image model table:

```
Schema::create('images', function (Blueprint $table){
            $table->increments('id');
            $table->string('url');
            $table->integer('imageable_id');
            $table->string('imageable_type');
            $table->timestamps();
        });
```

When i try to use CRUD in TicketCrudController (I got Exception that array needed, but string sent):

```
$this->crud->addField([    // Image
            'label' => 'ticket Image',
            'type' => 'image',
            'name' => 'images',
            'entity' => 'images',
            'attribute' => 'filename',
            'model' => "App\Models\Image",
            'upload' => true,
            'disk' => 'uploads',
            'morph' => true,
        ]);
```

i am getting error 

`Property [images] does not exist on this collection instance.`

any one please help me how i can use it

**Comment by umarakber** [comment]
i am also getting issue 
`Call to undefined method Illuminate\Database\Eloquent\Relations\MorphMany::sync()`

**Comment by davybaccaert** [comment]
@umarakber i had the same problem.

The issue sits in the src/PanelTraits/Create.php where on line 138 it tries to sync a 
morphMany (hasMany in essence) and that does not allow syncing. 

Replace 

```
$values = $data[$field['name']];
$model->{$field['name']}()->sync($values);
```

with 

`$model->save($data);`

works like a charm. 
No idea if its the right way to do it though.

**Comment by kiddtang** [comment]
similiar issue here... 
unitl i get rid of this line...
```
 $this->createRelations($item, $data);
```

my way is to override the store method without changing the vendor file...

```
    public function store()
    {
        $request = CRUD::getRequest();
        $request->request->set('user_id', backpack_user()->id);

        $this->crud->hasAccessOrFail('create');

        // execute the FormRequest authorization and validation, if one is required
        $request = $this->crud->validateRequest();

        // insert item in the db
        $data = $this->crud->getStrippedSaveRequest();
        $data = $this->crud->decodeJsonCastedAttributes($data);
        $data = $this->crud->compactFakeFields($data);

        // omit the n-n relationships when updating the eloquent item
        $model = new \App\Models\SomeModel();
        $nn_relationships = Arr::pluck($this->crud->getRelationFieldsWithPivot(), 'name');
        $item = $model->create(Arr::except($data, $nn_relationships));

        // if there are any relationships available, also sync those
        // $this->createRelations($item, $data);  <------- this line causing the issue
        $this->data['entry'] = $this->crud->entry = $item;

        // show a success message
        \Alert::success(trans('backpack::crud.insert_success'))->flash();

        // save the redirect choice for next time
        $this->crud->setSaveAction();

        $response = $this->crud->performSaveAction($item->getKey());

        return $response;
    }
```

**Comment by pxpm** [comment]
This is fixed in #3410 

It's not merged yet, still in review and polish phase, but is pretty much working, and some people are already using it. 

If you guys happen to give a try to the PR before it gets merged, please leave there a comment about your experience. If you find any bug it would be good to know before we merge it 😄 

I am going to close this. So far the best method to store those relations is by overriding the saving process in the CRUD. We hope to change that with the PR I mentioned.

That said I am going to close this, if you have any additional info in this regard please report on the PR thread.

Thanks for your patience,
Pedro
